pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -181,3 +197,104 @@ func validateExternalClusterSpec(cluster *cluster) error {
+	labels := controller.AppLabels(mgr.AppName, cluster.Namespace)
+
+	endpoints := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: cluster.Namespace,
+			Labels:    labels,
+		},
+		Subsets: []v1.EndpointSubset{
+			{
+				Addresses: cluster.Spec.Monitoring.ExternalMgrEndpoints,
+				Ports: []v1.EndpointPort{
+					{
+						Name:     serviceExternalMetricName,
+						Port:     int32(mgr.MetricsPort),
+						Protocol: v1.ProtocolTCP,
+					},
+				},
+			},
+		},
+	}
+
+	k8sutil.SetOwnerRef(&endpoints.ObjectMeta, &cluster.ownerRef)
+	return endpoints
+}
+
+func (c *ClusterController) configureExternalClusterMonitoring(cluster *cluster) error {
+	// Initialize manager object
+	manager := mgr.New(cluster.Info,
+		c.context,
+		c.namespacedName.Namespace,
+		c.rookImage,
+		cephv1.CephVersionSpec{},
+		rookv1.Placement{},
+		rookv1.Annotations{},
+		cephv1.NetworkSpec{},
+		cephv1.DashboardSpec{},
+		cluster.Spec.Monitoring, // ONLY IMPORTANT PART
+		cephv1.MgrSpec{},
+		v1.ResourceRequirements{},
+		"",
+		cluster.ownerRef,
+		cluster.Spec.DataDirHostPath,
+		cluster.Spec.SkipUpgradeChecks,
+	)
+
+	// Create external monitoring Endpoints
+	endpoint := createExternalMetricsEndpoints(cluster)
+	logger.Info("creating mgr external monitoring endpoints")
+	_, err := c.context.Clientset.CoreV1().Endpoints(c.namespacedName.Namespace).Create(endpoint)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "failed to create mgr endpoints")
+		}
+		logger.Info("mgr external metrics endpoints already exist")
+	} else {
+		logger.Info("mgr external metrics endpoints created")
+	}
+
+	// Create external monitoring Service
+	service := manager.MakeMetricsService(appName, serviceExternalMetricName)
+	logger.Info("creating mgr external monitoring service")
+	_, err = c.context.Clientset.CoreV1().Services(c.namespacedName.Namespace).Create(service)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "failed to create mgr service")
+		}
+		logger.Info("mgr external metrics service already exists")
+	} else {
+		logger.Info("mgr external metrics service created")
+	}
+
+	// Deploy external ServiceMonittor
+	logger.Info("creating external service monitor")
+	// servicemonitor takes some metadata from the service for easy mapping
+	err = manager.EnableServiceMonitor(service)
+	if err != nil {
+		logger.Errorf("failed to enable external service monitor. %v", err)
+	} else {
+		logger.Info("external service monitor created")
+	}
+
+	// namespace in which the prometheusRule should be deployed
+	// if left empty, it will be deployed in current namespace
+	namespace := cluster.Spec.Monitoring.RulesNamespace
+	if namespace == "" {
+		namespace = cluster.Namespace
+	}
+
+	logger.Info("creating external prometheus rule")
+	err = manager.DeployPrometheusRule(mgr.PrometheusRuleName, namespace)
+	if err != nil {
+		logger.Errorf("failed to create external prometheus rule. %v", err)
```

Comment:
No error for already exists?

Reply:
`DeployPrometheusRule()` does much more than that and the `IsAlreadyExists` is handled by `CreateOrUpdatePrometheusRule()`.